### PR TITLE
Added transaction api 147

### PR DESCRIPTION
**Pull request by artfred16**
**Added endpoints**
`GET api/transactions/{userId}` => list user transactions
`GET api/transactions/{userId}/{trackingNumber}` => show specific transaction (isolated to user)
`DELETE api/'transactions/{userId}/{trackingNumber}` => cancel specific transaction (isolated to user)

*Update transaction method for admin. `*NOTE: No route and guard yet.`

### CODE REVIEW

File: app/Models/Transaction.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    const TO_SHIP = 6;
+    const TO_RECEIVE = 7;
+
+    public static function boot() {
+        parent::boot();
+
+        self::creating(function($user)
+        {
+            // TODO: Create a tracking number generator
+            $user->tracking_number = Str::uuid();
```
Comment (kristher1619) [inline app/Models/Transaction.php]: I think uuid is already okay with tracking number? do you have any suggestions?

File: database/migrations/2020_08_23_125736_create_transactions_table.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     */
+    public function up()
+    {
+        Schema::create('transactions', function (Blueprint $table) {
+            $table->id();
+            $table->string('user_id');
+            $table->integer('transaction_type')->default(Transaction::DEBIT);
+            $table->double('amount');
+            $table->string('tracking_number');
+            $table->integer('status')->default(Transaction::PREPARING);
```
Comment (kristher1619) [inline database/migrations/2020_08_23_125736_create_transactions_table.php]: use unsignedInteger

File: routes/api.php [hunk 3]
```diff
@@ -22,3 +22,6 @@
 Route::get('cart/{userId}', "Api\CartController@index");
 Route::put('cart/{userId}', "Api\CartController@update");
 Route::delete('cart/{userId}', "Api\CartController@destroy");
+Route::get('transactions/{userId}', "Api\TransactionController@index");
+Route::get('transactions/{userId}/{trackingNumber}', "Api\TransactionController@show");
+Route::delete('transactions/{userId}/{trackingNumber}', "Api\TransactionController@destroy");
```
Comment (kristher1619) [inline routes/api.php]: lets use /{userId}/transactions/{trackingNumber}. I might also change the cart api routing

**Review by kristher1619 — CHANGES_REQUESTED** [review]

File: app/Models/Transaction.php (hunk 1 above)
Comment (artfred16) [reply app/Models/Transaction.php]: We should have a human readable tracking number,
- can be dictated in case of followup

**Review by kristher1619 — APPROVED** [review]

**Pull request merged**